website/docs/getting_started.html.markdown

```diff
@@ -0,0 +1,230 @@
+names.
+
+If you look at the [`google_compute_instance documentation`](/docs/providers/google/r/compute_instance.html),
+you'll see that `project` and `zone` (VM instances are a zonal resource) are
+listed as optional. When present in a resource's config block, these values will
+be used. If omitted, the provider defaults will be used instead.
+
+Add the following to your config file:
+
+```hcl
+resource "google_compute_instance" "vm_instance" {
+  name         = "terraform-instance"
+  machine_type = "f1-micro"
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
+    }
+  }
+
+  network_interface {
+    # A default network is created for all GCP projects
+    network       = "default"
+    access_config = {
+    }
+  }
+}
+```
+
+~> Note: Don't use `terraform apply` quite yet! You still need to add GCP
+credentials. If you want to try out provisioning your VM instance before
+continuing, follow the instructions in the "Adding credentials" section below.
+
+## Linking GCP resources
+
+Like this VM instance, nearly every GCP resource will have a `name` field. They
+are used as a short way to identify resources, and a resource's display name in
+the Cloud Console will be the one defined in the `name` field.
+
+When linking resources in a Terraform config though, you'll primarily want to
+use a different field, the `self_link` of a resource. Like `name`, nearly every
+resource has a `self_link`. They look like:
+
+```
+{{API base url}}/projects/{{your project}}/{{location type}}/{{location}}/{{resouce type}}/{{name}}
+```
+
+For example, the instance defined earlier in a project named `foo` will have
+the `self_link`:
+
+```
+https://www.googleapis.com/compute/v1/projects/foo/zones/us-central1-c/instances/terraform-instance
+```
+
+A resource's `self_link` is a unique reference to that resource. When
+linking two resources in Terraform, you can use Terraform interpolation to
+avoid typing out the self link! Let's use a `google_compute_network` to
+demonstrate.
+
+Add this block to your config:
+
+```hcl
+resource "google_compute_network" "vpc_network" {
+  name                    = "terraform-network"
+  auto_create_subnetworks = "true"
+}
+```
+
+This will create [VPC network resource](/docs/providers/google/r/compute_network.html)
+with a subnetwork in each region. Then, change the network of the
+`google_compute_instance` from the `"default"` network to the new network.
+
+```diff
+network_interface {
+-  # A default network is created for all GCP projects
+-  network = "default"
++  network = "${google_compute_network.vpc_network.self_link}"
+  access_config = {
+```
+
+This means that when we create the VM instance, it will use
+`"terraform-network"` instead of the default VPC network for the project. If you
+run `terraform plan`, you will see that `"terraform-instance"` depends on
+`"terraform-network"`.
+
+You can now run `terraform apply`! Except for one last thing...
```

Comment:
If you wanted to keep the tone less conversational (drier?) I might change to "Before you can run terraform apply you must add credentials"